sdc/rewrites/dataframe_constructor.py

```diff
@@ -0,0 +1,200 @@
+
+    _df_arg_list = ('data', 'index', 'columns', 'dtype', 'copy')
+
+    def __init__(self, pipeline):
+        super().__init__(pipeline)
+
+        self._reset()
+
+
+    def match(self, func_ir, block, typemap, calltypes):
+        self._reset()
+
+        self._block = block
+        self._func_ir = func_ir
+        self._calls_to_rewrite = set()
+
+        for stmt in find_operations(block=block, op_name='call'):
+            expr = stmt.value
+            fdef = guard(find_callname, func_ir, expr)
+            if fdef == self._pandas_dataframe:
+                args = get_call_parameters(call=expr, arg_names=self._df_arg_list)
+
+                if self._match_tuple_case(args, block, func_ir):
+                    self._calls_to_rewrite.add((stmt, self._extract_tuple_args, lambda *args: None))
+                elif self._match_dict_case(args, block, func_ir):
+                    self._calls_to_rewrite.add((stmt, self._extract_dict_args, self._remove_dict_var))
+                else:
+                    pass #TODO: throw error?
+
+        return len(self._calls_to_rewrite) > 0
+
+
+    def apply(self):
+        init_df_stmt = import_function(init_dataframe, self._block, self._func_ir)
+
+        for stmt, extract, cleanup in self._calls_to_rewrite:
+            args = get_call_parameters(call=stmt.value, arg_names=self._df_arg_list)
+
+            old_data = args['data']
+            old_columns = args['columns'] if 'columns' in args else None
+
+            args['data'], args['columns'] = extract(args, self._block, self._func_ir)
+
+            args['data'] = self._replace_with_arrays(args['data'], stmt, self._block, self._func_ir)
+
+            self._replace_call(stmt, init_df_stmt.target, args, self._block, self._func_ir)
+
+            cleanup(old_data, old_columns, self._block, self._func_ir)
+
+        return self._block
+
+
+    def _reset(self):
+        self._block = None
+        self._func_ir = None
+        self._calls_to_rewrite = None
+
+
+    @staticmethod
+    def _match_dict_case(args, block, func_ir):
+        if 'data' in args and is_dict(args['data'], block, func_ir):
+            return True
+
+        return False
+
+
+    @staticmethod
+    def _match_tuple_case(args, block, func_ir):
+        for name in ('data', 'columns'):
+            if name in args and is_tuple(args[name], block, func_ir):
+                return True
+
+        return False
+
+
+    @staticmethod
+    def _extract_tuple_args(args, block, func_ir):
+        data_args = get_tuple_items(args['data'], block, func_ir) if 'data' in args else None
+        columns_args = get_tuple_items(args['columns'], block, func_ir) if 'columns' in args else None
+
+        return data_args, columns_args
+
+
+    @staticmethod
+    def _extract_dict_args(args, block, func_ir):
+        dict_items = get_dict_items(args['data'], block, func_ir)
+
+        data_args = [item[1] for item in dict_items]
+        columns_args = [item[0] for item in dict_items]
```

Comment:
Probably you can use `dict_items.keys()` and `dict_items.values()` here.

Reply:
Unfortunately no. dict_items is list of tuples, not dict